server/log_rpc_server.go

```diff
@@ -504,7 +557,7 @@ func (t *TrillianLogRPCServer) getTreeAndHasher(ctx context.Context, treeID int6
 // InitLog initialises a freshly created Log by creating the first STH with size 0.
 func (t *TrillianLogRPCServer) InitLog(ctx context.Context, req *trillian.InitLogRequest) (*trillian.InitLogResponse, error) {
 	logID := req.LogId
-	tree, hasher, err := t.getTreeAndHasher(ctx, logID, false /* readonly */)
+	tree, hasher, err := t.getTreeAndHasher(ctx, logID, optsLogRead)
```

Comment:
write?

Reply:
Yes, fixed.

Actually, `InitLog` should accept `PREORDERED_LOG` in this line as well, but this requires `LogStorage.ReadWriteTransaction` refactoring similar to #1007. Left a TODO.